graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -153,12 +153,17 @@ def _construct_location_stack_entry(location, num_traverses):
 # - type: a GraphQL type object, like String or Integer, describing the type of that output value
 # - optional: boolean, whether the output is part of an optional traversal and
 #             could therefore have a value of null because it did not exist
-class OutputMetadata(namedtuple("OutputMetadata", ("type", "optional"))):
+# - folded: boolean, whether the output is within a fold scope
+class OutputMetadata(namedtuple("OutputMetadata", ("type", "optional", "folded"))):
```

Comment:
Since we're already editing this class in a non-trivial way, consider making it a new-style `NamedTuple` with proper typed fields. I think it would clean up the documentation comment in addition to adding the new functionality.